src/redux-enhancers/undoRedoReducerEnhancer.ts

```diff
@@ -212,6 +212,20 @@ const undoRedoReducerEnhancer: StoreEnhancer<any> =
         return newState
       }
 
+      // if an undoable action is of type importText, update inversePatches adding latest patch of actions which contains accurate patches for importText action
+      if (actionType === 'importText') {
+        // we do not want to update inversePatches  when importText runs initially for tutorial event
+        if (!newState.inversePatches.length) {
+          return newState
+        }
+        lastActionType = actionType
+        const inversePatch = compareWithOmit(newState as Index, state)
+        return {
+          ...newState,
+          inversePatches: [...state.inversePatches.slice(0, -1), addActionsToPatch(inversePatch, [actionType])],
+        }
```

Comment:
What happens if you simply remove lines 221–226? Then it falls into the normal case and gets added to `inversePatches` as usual. Seems to work for me.

(While I could turn out to be wrong here, I will note that it was my previous comment that led me to question the need for this code at all. Justifying nontrivial code is therefore useful not just for documentation purposes, but can also help catch redundancies.)

Reply:
Yep, you are correct! This bit of code is actually non trivial because the same logic is easliy handled at the end of function. I will simply just get rid of it! Thanks for the clarification :)